### PR TITLE
feat: example WebSocket

### LINKED ISSUE #84: WebSocket support

### PR DESCRIPTION
**Pull request by chesedo**
Adds a ws example

Close #84

### CODE REVIEW

File: api/Cargo.toml [hunk 1]
```diff
@@ -18,8 +18,8 @@ serde = "1.0.136"
 structopt = "0.3.26"
 sqlx = { version = "0.5", features = ["runtime-tokio-native-tls", "postgres"] }
 tokio = { version = "1.15.0", features = ["full"] }
-# not great, but the changes for hyper 0.14 and tokio 1 have not been merged on the main repo
-hyper-reverse-proxy = { git = "https://github.com/jtroo/hyper-reverse-proxy" }
+# not great, but waiting for WebSocket changes to be merged
+hyper-reverse-proxy = { git = "https://github.com/chesedo/hyper-reverse-proxy", branch = "master" }
```
Comment (brokad) [inline api/Cargo.toml:22]: Seems this was [merged upstream](https://github.com/felipenoris/hyper-reverse-proxy/pull/35).

File: examples/axum/websocket/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+#[shuttle_service::main]
+async fn main() -> Result<SyncWrapper<Router>, shuttle_service::Error> {
+    let (tx, rx) = watch::channel(Message::Text("{}".to_string()));
+
+    let state = Arc::new(Mutex::new(State {
+        clients_count: 0,
+        rx,
+    }));
+
+    // Spawn a thread to continually chech the status of the api
```
Comment (brokad) [inline examples/axum/websocket/src/lib.rs:47]:
```suggestion
    // Spawn a thread to continually check the status of the api
```

File: examples/axum/websocket/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    time::sleep,
+};
+
+struct State {
+    clients_count: usize,
+    rx: watch::Receiver<Message>,
+}
+
+const PAUSE_SECS: u64 = 15;
+const STATUS_URI: &str = "https://api.shuttle.rs/status";
```
Comment (brokad) [inline examples/axum/websocket/src/lib.rs:29]: Very meta :smile:

**Review by brokad — APPROVED** [review]

File: api/Cargo.toml (hunk 1 above)
Comment (chesedo) [reply api/Cargo.toml:22]: Yip, it's just [not released yet](https://github.com/felipenoris/hyper-reverse-proxy/releases) :disappointed:

**Pull request merged**